### PR TITLE
contributing: update minimum supported Rust policy

### PR DESCRIPTION
**Pull request by davidhewitt**
Based off discussion in #1420 

cc @kngwyu @birkenfeld

### CODE REVIEW

File: Contributing.md [hunk 1]
```diff
@@ -53,15 +53,25 @@ Formatting, linting and tests are checked for all Rust and Python code. In addit
 
 Tests run with all supported Python versions with the latest stable Rust compiler, as well as for Python 3.9 with the minimum supported Rust version.
 
-### Minimum supported Rust version
+## Python and Rust version support policy
+
+PyO3 aims to keep sufficient compatibility to make packaging Python extensions built with PyO3 feasible on most common package managers.
+
+To keep package maintainer's lives simpler, PyO3 will commit, wherever possible, to only adjust minimum supported Rust and Python versions at the same time. This release will come roughly once per year, after the oldest supported Python version reaches its end-of-life. (Check https://endoflife.date/python for a clear timetable on these.)
```
Comment (birkenfeld) [inline Contributing.md:60]:
```suggestion
To keep package maintainers' lives simpler, PyO3 will commit, wherever possible, to only adjust minimum supported Rust and Python versions at the same time. This bump will come in a new `0.x` release, roughly once per year, after the oldest supported Python version reaches its end-of-life. (Check https://endoflife.date/python for a clear timetable on these.)
```

**Review by birkenfeld — APPROVED** [review]
just a few nits...

File: Contributing.md [hunk 2]
```diff
@@ -53,15 +53,25 @@ Formatting, linting and tests are checked for all Rust and Python code. In addit
...
+
+Below are guidelines on what compatibility all PRs are expected to deliver for each language.
+
+## Python
```
Comment (kngwyu) [inline Contributing.md:64]: You meant:
```suggestion
### Python
```
?

**Review by kngwyu — APPROVED** [review]
Nice, thank you!

File: Contributing.md (hunk 2 above)
Comment (davidhewitt) [reply Contributing.md:64]: 👍

File: Contributing.md (hunk 1 above)
Comment (davidhewitt) [reply Contributing.md:60]: Hmm do you think `in a new 0.x release` implies we won't make 0.x releases at other times in the year?

Comment (birkenfeld) [reply Contributing.md:60]: Right, it might be read that way.  maybe `will only come in a 0.x release`?

File: Contributing.md [hunk 3]
```diff
@@ -53,15 +53,25 @@ Formatting, linting and tests are checked for all Rust and Python code. In addit
...
+
+To keep package maintainer's lives simpler, PyO3 will commit, wherever possible, to only adjust minimum supported Rust and Python versions at the same time. This release will come roughly once per year, after the oldest supported Python version reaches its end-of-life. (Check https://endoflife.date/python for a clear timetable on these.)
+
+Below are guidelines on what compatibility all PRs are expected to deliver for each language.
+
+### Python
+
+PyO3 supports all oficially supported Python versions, as well as the latest PyPy3 release. All of these versions are tested in CI.
+
+## Rust
```
Comment (kngwyu) [inline Contributing.md:68]: Same as above,
```suggestion
### Rust
```

Comment (davidhewitt) [reply Contributing.md:68]: Thanks!

File: Contributing.md (hunk 1 above)
Comment (davidhewitt) [reply Contributing.md:60]: I like that, let's go for that.

**Pull request merged**